### PR TITLE
feature: scanning broken link

### PR DESCRIPTION
**Pull request by shaowenchen**
Signed-off-by: shaowenchen <mail@chenshaowen.com>

**What type of PR is this?**

 /kind feature

### CODE REVIEW

**Comment by shaowenchen** [comment]
@kubesphere/sig-docs  Please pick the pr to your repo and fix the broken links

File: .github/workflows/checklink.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+jobs:
+  markdown-link-check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - uses: gaurav-nelson/github-action-markdown-link-check@v1
+      with:
+        use-quiet-mode: 'yes'
+        config-file: '.github/workflows/checklink_config.json'
```
Comment (LinuxSuRen) [inline .github/workflows/checklink.yaml:13]: How about put configuration files into `.github`? And keep only put workflow files into `.github/workflows`.

File: .github/workflows/checklink.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+name: Check Markdown links
+
+on: push
```
Comment (LinuxSuRen) [inline .github/workflows/checklink.yaml:3]: It's a little bit late once the broken links get merged. How about running the link on each pull request.

File: .github/workflows/checklink.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+jobs:
+  markdown-link-check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - uses: gaurav-nelson/github-action-markdown-link-check@v1
+      with:
+        use-quiet-mode: 'yes'
+        config-file: '.github/workflows/checklink_config.json'
+        max-depth: 3
```
Comment (LinuxSuRen) [inline .github/workflows/checklink.yaml:14]: Why three?

**Review by LinuxSuRen — CHANGES_REQUESTED** [review]
I left some comments. Please take look at it.

File: .github/workflows/checklink.yaml (hunk 1 above)
Comment (shaowenchen) [reply .github/workflows/checklink.yaml:13]: More repositories are placed in the same directory in [Real-life usage samples](https://github.com/gaurav-nelson/github-action-markdown-link-check#real-life-usage-samples), which is just two styles.

File: .github/workflows/checklink.yaml (hunk 3 above)
Comment (shaowenchen) [reply .github/workflows/checklink.yaml:14]: It is to ignore the directory of vendor.

File: .github/workflows/checklink.yaml (hunk 2 above)
Comment (shaowenchen) [reply .github/workflows/checklink.yaml:3]: fixed

File: .github/workflows/checklink.yaml (hunk 1 above)
Comment (LinuxSuRen) [reply .github/workflows/checklink.yaml:13]: I see. This is not a block comment. They're both ok.

File: .github/workflows/checklink.yaml (hunk 3 above)
Comment (LinuxSuRen) [reply .github/workflows/checklink.yaml:14]: How about only check the `doc` and other necessary dir?

Comment (LinuxSuRen) [reply .github/workflows/checklink.yaml:14]: How about only check the `doc` and other necessary dir?

**Pull request closed without merging**